RUFAS/output_manager.py

```diff
@@ -549,25 +549,72 @@ def _dict_to_csv_column_list(self, variable_name: str, data_dict: Dict[str, List
-            else:
-                column_title = f"{variable_name}"
-                column_list.append(pd.Series(data_list, dtype=object, name=column_title))
+        units = data_dict["info_maps"][0]["units"] if data_dict.get("info_maps", []) else None
+        data_list = data_dict["values"]
+        if data_list and isinstance(data_list[0], dict):
+            csv_column_lists: Dict[str, List[Any]] = {subkey: [] for item in data_list for subkey in item.keys()}
+            for nested_dictionary in data_list:
+                for subkey, value in nested_dictionary.items():
+                    csv_column_lists[subkey].append(value)
+
+            for subkey in csv_column_lists.keys():
+                column_title = f"{variable_name}.{subkey}{self._get_units_substr(variable_name, units, subkey)}"
+                column_list.append(pd.Series(csv_column_lists[subkey], dtype=object, name=column_title))
+        else:
+            column_title = f"{variable_name}{self._get_units_substr(variable_name, units)}"
+            column_list.append(pd.Series(data_list, dtype=object, name=column_title))
 
         return column_list
 
+    def _get_units_substr(
+        self, variable_name: str, units: str | Dict[str, str] | None, subkey: str | None = None
+    ) -> str:
+        """Get the units substring for a column title.
+
+        Parameters
+        ----------
+        variable_name : str
+            The name of the variable or group of variables associated with the units.
+        units : str | Dict[str, str] | None
+            The units associated with the data.
+        subkey : str | None, optional
+            The subkey to retrieve the units for, if units is a dictionary. Default is None.
+
+        Returns
+        -------
+        str
+            The formatted units substring for the column title.
+
+        Examples
+        --------
+        >>> output_manager = OutputManager()
+        >>> output_manager._get_units_substr("temperature", "C")
+        '_(C)'
+        >>> output_manager._get_units_substr("velocity", {"magnitude": "m/s", "direction": "degrees"}, "magnitude")
+        '_(m/s)'
+        >>> output_manager._get_units_substr("velocity", {"magnitude": "m/s", "direction": "degrees"}, "direction")
+        '_(degrees)'
+        """
+
+        if not isinstance(units, dict):
+            return f"_({units})" if units else ""
+
+        if subkey not in units:
+            if subkey is not None:
+                self.add_error(
+                    "units_key_error",
+                    f"Key '{subkey}' not found in the units dictionary for variable '{variable_name}'.",
+                    info_map={
+                        "class": self.__class__.__name__,
+                        "function": self._get_units_substr.__name__,
+                    },
+                )
+            return ""
+
+        return f"_({units[subkey]})"
```

Comment:
```suggestion
        return f" ({units[subkey]})"
```
Instead of appending with an underscore, I think it would be cleaner to just the units enclosed in parentheses. This makes the column a little more readable, especially for non-devs who don't spend all their reading and writing things in snake case.

Reply:
Okay. I will add a whitespace between the variable name and the units.